### PR TITLE
Added support for SAS policy for Azure topic plugin.

### PR DESCRIPTION
**Pull request by addp009**

### CODE REVIEW

**Comment by xiaomi7732** [comment]
@addp009,
Thanks for your contribution. It is great and clean implementation that makes SAS working.

Do you think it possible to make the code backward compatible? So that existing user using ACS without changing their configuration won't be broken?

File: Logstash/logstash-input-azuretopic/lib/logstash/inputs/azuretopic.rb [hunk 1]
```diff
@@ -27,16 +28,21 @@ def initialize(*args)
...
+
+    @azure_service_bus = Azure::ServiceBus::ServiceBusService.new(sb_host, { signer: signer})
   end # def register
 
   def process(output_queue)
     message = @azure_service_bus.receive_subscription_message(@topic ,@subscription, { :peek_lock => true, :timeout => 1 } )
     if message
       codec.decode(message.body) do |event|
         decorate(event)
+        event.set("type", @topic)
```
Comment (xiaomi7732) [inline Logstash/logstash-input-azuretopic/lib/logstash/inputs/azuretopic.rb]: Could you please explain why do we set this hard-coded property of type to the event? What if there's already property named 'type' in the log?

Comment (addp009) [inline Logstash/logstash-input-azuretopic/lib/logstash/inputs/azuretopic.rb]: Sorry, this was a irrelevant change due to user error. ;)

**Comment by addp009** [comment]
Added backwards compatibility to ACS, and updated docs to indicate how to use SAS policy.

**Comment by xiaomi7732** [comment]
@addp009,
Thanks for the quick turn-around. Good job!

**Pull request merged**